indexer-balances/src/models/mod.rs

```diff
@@ -30,6 +34,45 @@ pub async fn chunked_insert<T: SqlxMethods + std::fmt::Debug>(
     try_join_all(futures).await.map(|_| ())
 }
 
+pub(crate) async fn select_one_retry_or_panic(
```

Comment:
I tried to make this generic but think I ended up with something more confusing. The intention was to provide any constructed query and this method handles execution/retires.

Does it make sense?